### PR TITLE
Closes #1336: Migrate old runs & clean up code for status displays

### LINKED ISSUE #1336: Remove status_display_pre_result_monitor and status_display_pre_run_stages
after migrating old pipeline_runs to the new data model

### PR DESCRIPTION
**Pull request by cdebourcy**
1. Run `rake migrate_run_status`
2. See if status displays in samples list are still correct with the pruned samples controller code

### CODE REVIEW

File: lib/tasks/migrate_run_status.rake [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      OutputState.create(pipeline_run_id: pr.id, output: output, state: PipelineRun::STATUS_LOADED)
+    end
+  elsif %w[FAILED ERROR].include?(pr.job_status)
+    pr.create_output_states
+    pr.update(results_finalized: PipelineRun::FINALIZED_FAIL)
+  else
+    pr.create_output_states
+    pr.update(results_finalized: PipelineRun::IN_PROGRESS) # shouldn't be the case for any of those old runs
+  end
+  pr.create_run_stages
```
Comment (yunfangjuan) [inline lib/tasks/migrate_run_status.rake]: I will probably move this to first line of the function.

File: lib/tasks/migrate_run_status.rake [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+def migrate_pre_run_stages(pr)
+  target_outputs = %w[ercc_counts taxon_counts taxon_byteranges]
+  if %w[CHECKED SUCCEEDED].include?(pr.job_status)
+    pr.update(results_finalized: PipelineRun::FINALIZED_SUCCESS)
+    target_outputs.each do |output|
+      OutputState.create(pipeline_run_id: pr.id, output: output, state: PipelineRun::STATUS_LOADED)
+    end
+  elsif %w[FAILED ERROR].include?(pr.job_status)
+    pr.create_output_states
```
Comment (yunfangjuan) [inline lib/tasks/migrate_run_status.rake]: I will porbably do pr.create_output_states in the beginning of the function and handle the update afterwards to make it cleaner.

Comment (cdebourcy) [reply lib/tasks/migrate_run_status.rake]: ok

File: lib/tasks/migrate_run_status.rake (hunk 1 above)
Comment (cdebourcy) [reply lib/tasks/migrate_run_status.rake]: ok

**Pull request merged**